x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportUpdateInferenceModelAction.java

```diff
@@ -293,12 +293,23 @@ private boolean isInClusterService(String name) {
         return List.of(ElasticsearchInternalService.NAME, ElasticsearchInternalService.OLD_ELSER_SERVICE_NAME).contains(name);
     }
 
-    private void throwIfTrainedModelDoesntExist(UpdateInferenceModelAction.Request request) throws ElasticsearchStatusException {
-        var assignments = TrainedModelAssignmentUtils.modelAssignments(request.getInferenceEntityId(), clusterService.state());
+    private String getDeploymentIdForInClusterEndpoint(String inferenceEntityId, Model model) {
+        var serviceSettings = model.getConfigurations().getServiceSettings();
+        if (serviceSettings instanceof ElasticsearchInternalServiceSettings) {
+            var deploymentId = ((ElasticsearchInternalServiceSettings) serviceSettings).getDeploymentId();
+            if (deploymentId != null) {
+                return deploymentId;
+            }
+        }
+        return inferenceEntityId;
```

Comment:
`ElasticsearchInternalModel::mlNodeDeploymentId()` contains this same logic:

```suggestion
        if (model instanceof ElasticsearchInternalModel esModel) {
            return esModel.mlNodeDeploymentId();
        } else {
            // should never happen
            throw new IllegalStateException(
                Strings.format(
                    "Cannot update inference endpoint [%s]. Class [%s] is not an Elasticsearch internal model",
                    model.getInferenceEntityId(),
                    model.getClass().getSimpleName()
                )
            );
        }
```

Reply:
Good catch, I've updated this to reuse the existing logic.